crates/sqlexec/src/planner/physical_plan/describe_table.rs

```diff
@@ -0,0 +1,211 @@
+    ) -> crate::errors::Result<Self, protogen::ProtoConvError> {
+        Ok(Self {
+            tbl_reference: proto.tbl_reference.unwrap().try_into().unwrap(),
+        })
+    }
+}
+
+impl ExecutionPlan for DescribeTableExec {
+    fn as_any(&self) -> &dyn Any {
+        self
+    }
+
+    fn schema(&self) -> SchemaRef {
+        DESCRIBE_TABLE_SCHEMA.clone()
+    }
+
+    fn output_partitioning(&self) -> Partitioning {
+        Partitioning::UnknownPartitioning(1)
+    }
+
+    fn output_ordering(&self) -> Option<&[PhysicalSortExpr]> {
+        None
+    }
+
+    fn children(&self) -> Vec<Arc<dyn ExecutionPlan>> {
+        Vec::new()
+    }
+
+    fn with_new_children(
+        self: Arc<Self>,
+        _: Vec<Arc<dyn ExecutionPlan>>,
+    ) -> Result<Arc<dyn ExecutionPlan>> {
+        Err(DataFusionError::Plan(
+            "Cannot change children for DescribeTableExec".to_string(),
+        ))
+    }
+
+    fn execute(&self, _: usize, context: Arc<TaskContext>) -> Result<SendableRecordBatchStream> {
+        let mutator = context
+            .session_config()
+            .get_extension::<CatalogMutator>()
+            .expect("context should have catalog mutator");
+
+        let catalog = context
+            .session_config()
+            .get_extension::<SessionCatalog>()
+            .expect("context should have catalog");
+
+        let temp_catalog = context.session_config().get_extension::<TempCatalog>();
+
+        let vars = context
+            .session_config()
+            .options()
+            .extensions
+            .get::<SessionVars>()
+            .expect("context should have SessionVars");
+
+        let stream = stream::once(describe_table(
+            self.tbl_reference.clone(),
+            mutator,
+            catalog,
+            temp_catalog,
+            Arc::new(vars.clone()),
+        ));
+
+        Ok(Box::pin(RecordBatchStreamAdapter::new(
+            self.schema(),
+            stream,
+        )))
+    }
+
+    fn statistics(&self) -> Statistics {
+        Statistics::default()
+    }
+}
+
+async fn describe_table(
+    tbl_ref: OwnedFullObjectReference,
+    mutator: Arc<CatalogMutator>,
+    catalog: Arc<SessionCatalog>,
+    temp_catalog: Option<Arc<TempCatalog>>,
+    vars: Arc<SessionVars>,
+) -> Result<RecordBatch> {
+    let metastore_client = mutator.get_metastore_client();
+    let mut catalog = catalog.as_ref().clone();
+    let schema_search_path = vars.implicit_search_path();
+    let temp_catalog = temp_catalog.unwrap_or_default();
+
+    catalog
+        .maybe_refresh_state(metastore_client, false)
+        .await
+        .unwrap();
+
+    let resolver = EntryResolver {
+        catalog: &catalog,
+        temp_objects: temp_catalog,
+        schema_search_path,
+    };
+
+    let ent = resolver
+        .resolve_entry_from_reference(tbl_ref.into())
+        .and_then(ResolvedEntry::try_into_table_entry)
+        .expect("entry should be checked during planning");
```

Comment:
I'd return an error here since we don't have a guarantee that the table hasn't been dropped between planning and execution.